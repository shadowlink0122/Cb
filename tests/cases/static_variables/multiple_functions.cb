// 複数関数でのstatic変数スコープテスト
int func1(){
    static int value = 10;
    value += 5;
    return value;
}

int func2(){
    static int value = 100;
    value -= 10;
    return value;
}

int func3(){
    static int value = 1;
    value *= 2;
    return value;
}

int main(){
    // 各関数のstatic変数は独立している
    println(func1()); // 15
    println(func2()); // 90
    println(func3()); // 2
    
    println(func1()); // 20
    println(func2()); // 80
    println(func3()); // 4
    
    println(func1()); // 25
    println(func2()); // 70
    println(func3()); // 8
    
    return 0;
}
