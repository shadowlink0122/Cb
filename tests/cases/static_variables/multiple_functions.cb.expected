15
90
2
20
80
4
25
70
8
